ecosystem/platform/server/app/javascript/controllers/minted_nft_controller.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+import { Controller } from "./controller";
+import type { Types } from "aptos";
+
+// Connects to data-controller="minted-nft"
+export default class extends Controller {
+  static values = {
+    transactionHash: String,
+    apiUrl: String,
+  };
+
+  static targets = ["dateMinted", "mintNumber", "image"];
+
+  declare readonly dateMintedTarget: HTMLElement;
+  declare readonly mintNumberTarget: HTMLElement;
+  declare readonly imageTarget: HTMLImageElement;
+
+  declare readonly transactionHashValue: string;
+  declare readonly apiUrlValue: string;
+
+  connect() {
+    this.fetchNftInfo();
+  }
+
+  async fetchNftInfo() {
+    const transactionUrl = [
+      this.apiUrlValue,
+      'transactions',
+      'by_hash',
+      this.transactionHashValue,
+    ].join('/');
+    const response = await fetch(transactionUrl);
+    const transaction: Types.OnChainTransaction = await response.json();
+
+    if (!('timestamp' in transaction && 'events' in transaction)) return;
+
+    const createEvent = transaction.events.find(event =>
+      event.type === '0x3::token::CreateTokenDataEvent');
+
+    if (createEvent == null) return;
+
+    const dateMinted = new Date(parseInt(transaction.timestamp) / 1000);
```

Comment:
should we have `new Date(parseInt(transaction.timestamp) / 1000)` in a global functions file?

Reply:
If we end up needing it in multiple places, yes, but for now, no.